### PR TITLE
VIVO-1752: Initialize Solr Core on startup

### PR DESCRIPTION
**Pull request by wwelling**
[VIVO-1752](https://jira.lyrasis.org/browse/VIVO-1752)

@VIVO-project/vivo-committers

### CODE REVIEW

File: api/src/main/java/edu/cornell/mannlib/vitro/webapp/searchengine/solr/SolrSearchEngine.java [hunk 1]
```diff
@@ -199,13 +233,97 @@ public int documentCount() throws SearchEngineException {
...
+			.map(NamedList.class::cast)
+			.map(status -> status.get("name"))
+			.isPresent();
+	}
+
+	private void initializeCore(String contextPath) throws SolrServerException, IOException {
+		Path solrConfPath = Paths.get(contextPath + File.separator + "solr");
+		File solrConf = solrConfPath.toFile();
+
+		solrConf.setReadable(true, false);
```
Comment (awoods) [inline api/src/main/java/edu/cornell/mannlib/vitro/webapp/searchengine/solr/SolrSearchEngine.java:292]: These are both returning `false`... not surprisingly.

Comment (awoods) [reply api/src/main/java/edu/cornell/mannlib/vitro/webapp/searchengine/solr/SolrSearchEngine.java:292]: We may need to document a prerequisite of setting permissions on the Solr directories before start-up.

Comment (awoods) [reply api/src/main/java/edu/cornell/mannlib/vitro/webapp/searchengine/solr/SolrSearchEngine.java:292]: Both users need to be able to write to the core directories: 
- solr user
- tomcat user

Comment (wwelling) [reply api/src/main/java/edu/cornell/mannlib/vitro/webapp/searchengine/solr/SolrSearchEngine.java:292]: I tried changing the owner and that resulted in AccessDeniedException. I am open to ideas with permission issues.

Comment (wwelling) [reply api/src/main/java/edu/cornell/mannlib/vitro/webapp/searchengine/solr/SolrSearchEngine.java:292]: At the point of requiring changing directory permissions, it is not much more to copy the conf directory and create the core.

**Comment by awoods** [comment]
@wwelling : we can probably close this PR.

**Pull request closed without merging**